File: backend/app/schemas/category.py
from datetime import datetime
from pydantic import BaseModel, ConfigDict


class CategoriaCreate(BaseModel):
    nombre: str


class CategoriaUpdate(BaseModel):
    nombre: str


class CategoriaResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    nombre: str
    created_at: datetime
    updated_at: datetime

File: frontend/src/modules/categories/types/category.ts
export type CategoriaEstado = 'activo' | 'inactivo'

export interface Categoria {
  id: number
  nombre: string
  created_at: string
  updated_at: string
}

export interface CategoriaFilter {
  q?: string
}

export interface CategoriaCreate {
  nombre: string
}

export interface CategoriaUpdate {
  nombre: string
}
